R/rextendr_document.R

```diff
@@ -0,0 +1,23 @@
+#' This function is a wrapper for `devtools::document`.
+#' It executes `extendr`-specific routine before calling `devtools::document`,
+#' ensuring Rust code is recompiled when necessary and appropriate R wrappers are generated.
+#'
+#' @param pkg String scalar pointing to the package root.
+#' @param quiet Logical scalar indicating whether the output should be quiet (`TRUE`)
+#'   or verbose (`FALSE`).
+#' @param roclets Additional roclets from `roxygen2` that are passed into
+#'   `roxygen2::document`.
+#' @export
+document <- function(pkg = ".", quiet = FALSE, roclets = NULL) {
+  try_save_all()
+  needs_compilation <- needs_compilation(pkg, quiet) || pkgbuild::needs_compile(pkg)
+
+  pkgload::load_all(
+    path = pkg,
+    compile = needs_compilation,
+    quiet = quiet
+  )
```

Comment:
Note that we don't need this after #64 gets merged.

Reply:
That's the argument for me to merge #64 first.